src/cswinrt/code_writers.h

```diff
@@ -7640,26 +7653,203 @@ return false;
+            iid: in %ReferenceImpl.IID,
+            wrapperFlags: out wrapperFlags);
+
+        return new %(valueReference.Invoke);
+    }
+}
+)",
+            type.TypeName(),
+            type.TypeName(),
+            bind<write_type_name>(type, typedef_name_type::Projected, false)
+        );
+    }
+
+
+    void write_delegate_com_wrappers_marshaller_attribute_impl(writer& w, TypeDef const& type)
+    {
+        auto name = type.TypeName();
+        w.write(
+            R"(internal sealed unsafe class %ComWrappersMarshallerAttribute : WindowsRuntimeComWrappersMarshallerAttribute
+{
+    public override void* GetOrCreateComInterfaceForObject(object value)
+    {
+        return WindowsRuntimeMarshal.GetOrCreateComInterfaceForObject(value, CreateComInterfaceFlags.TrackerSupport);
+    }
+
+    public override ComInterfaceEntry* ComputeVtables(out int count)
+    {
+        count = sizeof(ReferenceInterfaceEntries) / sizeof(ComInterfaceEntry);
+        return (ComInterfaceEntry*)Unsafe.AsPointer(in %InterfaceEntriesImpl.Entries);
+    }
+}
+
+)", name, name);
+    }
+
+    void write_delegate_marshaller(writer& w, TypeDef const& type)
+    {
+        if (is_exclusive_to(type))
+        {
+            return;
+        }
+
+        auto projected_type = w.write_temp("%", bind<write_type_name>(type, typedef_name_type::Projected, false));
+        w.write(R"(
+[global::System.ComponentModel.EditorBrowsable(global::System.ComponentModel.EditorBrowsableState.Never)]
+public static unsafe class %Marshaller
+{
+    public static WindowsRuntimeObjectReferenceValue ConvertToUnmanaged(% value)
+    {
+        return WindowsRuntimeDelegateMarshaller.ConvertToUnmanaged(value, in %ReferenceImpl.IID);
+    }
+    #nullable enable
+    public static %? ConvertToManaged(void* value)
+    {
+        return (%?)WindowsRuntimeDelegateMarshaller.ConvertToManaged<%ComWrappersCallback>(value);
+    }
+    #nullable disable
+}
+)",
+            type.TypeName(),
+            bind<write_type_name>(type, typedef_name_type::Projected, false),
+            type.TypeName(),
+            projected_type,
+            projected_type,
+            type.TypeName()
+        );
+    }
+
+    void write_native_delegate(writer& w, TypeDef const& type)
+    {
+        auto method = get_delegate_invoke(type);
+        method_signature signature{ method };
+        auto generic_abi_types = get_generic_abi_types(w, signature);
+        bool have_generic_params = std::find_if(generic_abi_types.begin(), generic_abi_types.end(),
+            [](auto&& pair) { return !pair.second.empty(); }) != generic_abi_types.end();
+
+        w.write(R"(
+public static unsafe class %NativeDelegate
+{
+    public static % Invoke(this WindowsRuntimeObjectReference objectReference%%)
+    {
+        using WindowsRuntimeObjectReferenceValue ThisValue = objectReference.AsValue();
+        void* ThisPtr = ThisValue.GetThisPtrUnsafe();
+        var abiInvoke = ((%Vftbl*)ThisPtr)->Invoke;
```

Comment:
Nit: for consistency, name the value `objectValue` (we do the same in the runtime). Also use lowercase for locals, and leave a blank line. Lastly, imho it's cleaner to invoke in one line, also for consistency. We do the same everywhere else.
```suggestion
        using WindowsRuntimeObjectReferenceValue objectValue = objectReference.AsValue();

        void* thisPtr = ThisValue.GetThisPtrUnsafe();

        int hresult = ((%Vftbl*)ThisPtr)->Invoke(...)
```

Reply:
Agree with the lower case for local but write_abi_method_call_marshalers is using uppercase ThisPtr. If I were to change this, I am worried it will revibrate across all the class or interfaces abi imlementations. 

Let's do this fix another time.